### PR TITLE
Simplify CLI download

### LINKED ISSUE #334: Simplify downloads for headless servers
When trying to download from ore on a headless server using curl -O -J -L https://ore.spongepowered.org/Nucleus/Nucleus/versions/1.2.2-S7.0/download
it's provides a download link in a readme file involving tokens see below with a warning that it hasn't been reviewd
curl -O -J -L -d -X "https://ore.spongepowered.org/Nucleus/Nucleus/versions/1.2.2-S7.0/confirm?downloadType=0&token=zxzxzxzxzx&csrfToken=zzxzxzxzxzxz"

Can we remame the file from `README.txt` to something like `<plugin-slug>-<channel>-<version>-README.txt` and add support for Wget maybe with warning to use at your own risk

### PR DESCRIPTION
**Pull request by Faithcaio**
The API Endpoint
`GET     /api/projects/:pluginId/versions/:name/download`
`controllers.project.Versions.downloadJarById`
with token could never actually download the file because the warning was not getting confirmed.

Calling the Endpoint with a valid token will now automatically confirm it and download the requested file.

Attempting to download a unreviewed file without valid token returns the following Json:
```json
{
    "message": [
        "This version has not been reviewed by our moderation staff and may not be safe for download.",
        "  Disclaimer: We disclaim all responsibility for any harm to your server or system should you choose not to heed this   warning.",
        "  Please POST to the attached link to acknowledge this disclaimer and continue to the download."
    ],
    "post": "https://ore.spongepowered.org/<author>/<pluginname>/versions/<version>/confirm?downloadType=1&token=<token>",
    "url": "https://ore.spongepowered.org/api/projects/<pluginid>/versions/<version>/download?token=<token>",
    "token": "<token>"
}
```
"post": old url with token - continues to work with curl
"url": new url with token - works with everything
"token": just the token

fixes https://github.com/SpongePowered/Ore/issues/334

tl;dr:

To download a reviewed file run:
`wget https://ore.spongepowered.org/api/projects/<pluginId>/versions/<pluginVersion>/download`

To download a unreviewed file run:
`wget https://ore.spongepowered.org/api/projects/<pluginId>/versions/<pluginVersion>/download`
And again with the token:
`wget https://ore.spongepowered.org/api/projects/<pluginId>/versions/<pluginVersion>/download?token=<token>`

### CODE REVIEW

**Review by felixoi — APPROVED** [review]

File: app/controllers/project/Versions.scala [hunk 1]
```diff
@@ -527,25 +517,40 @@ class Versions @Inject()(stats: StatTracker,
...
-              self.confirmDownload(author, slug, target, Some(dlType.id), token).absoluteURL(),
-              CSRF.getToken.get.value) + "\n")
-              .withHeaders("Content-Disposition" -> "inline; filename=\"README.txt\""))
-          } else if (api.getOrElse(false)) {
-            Future.successful(MultipleChoices(Json.obj(
-              "message" -> this.messagesApi("version.download.confirm.body.api").split('\n'),
-              "post" -> helper.CSRF(
-                self.confirmDownload(author, slug, target, Some(dlType.id), token)).absoluteURL())))
+          if (api.getOrElse(false)) {
+            warning.map(warn => {
```
Comment (Katrix) [inline app/controllers/project/Versions.scala]: Only use curly braces for this function

File: app/controllers/project/Versions.scala [hunk 2]
```diff
@@ -718,8 +733,16 @@ class Versions @Inject()(stats: StatTracker,
       implicit val data = request.data
       implicit val p = data.project
       implicit val r = request.request
-      withVersionAsync(versionString)(version =>
-        sendJar(data.project, version, token, api = true))
+      withVersionAsync(versionString)(version => {
```
Comment (Katrix) [inline app/controllers/project/Versions.scala]: Only use curly braces for this function

File: app/controllers/project/Versions.scala [hunk 3]
```diff
@@ -527,25 +517,40 @@ class Versions @Inject()(stats: StatTracker,
...
+                "token" -> token)
+              )
+            })
           } else {
-            for {
-              warn <- warning
-              nonReviewed <- version.channel.map(_.isNonReviewed)
-            } yield MultipleChoices(views.unsafeDownload(data.project, version, nonReviewed , dlType, token)).withCookies(warn.cookie)
+            val userAgent = request.headers.get("User-Agent")
+            var curl: Boolean = false
```
Comment (Katrix) [inline app/controllers/project/Versions.scala]: Do we need the types annotations for these?

File: app/controllers/project/Versions.scala [hunk 4]
```diff
@@ -527,25 +517,40 @@ class Versions @Inject()(stats: StatTracker,
...
+            var wget: Boolean = false
+            if (userAgent.isDefined) {
```
Comment (Katrix) [inline app/controllers/project/Versions.scala]: Changing this to a foreach would be nice

File: app/controllers/project/Versions.scala (hunk 3 above)
Comment (Faithcaio) [reply app/controllers/project/Versions.scala]: Just moved the code from above.
But its gone now.

Also we might want to remove the user agent checking entirely.
Using CLI should be api = true anyways.
